Signal/src/ViewControllers/HomeView/ConversationSearchViewController.swift

```diff
@@ -0,0 +1,244 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+@objc
+class ConversationSearchViewController: UITableViewController {
+
+    var searchResultSet: SearchResultSet = SearchResultSet.empty
+
+    var uiDatabaseConnection: YapDatabaseConnection {
+        // TODO do we want to respond to YapDBModified? Might be hard when there's lots of search results, for only marginal value
+        return OWSPrimaryStorage.shared().uiDatabaseConnection
+    }
+
+    var searcher: ConversationSearcher {
+        return ConversationSearcher.shared
+    }
+
+    enum SearchSection: Int {
+        case conversations = 0
+        case contacts = 1
+        case messages = 2
```

Comment:
Should we have an additional "no matches" section we only show if there are no matches?

Reply:
There is an existing card for "No results". If adding another section proves to be the best way to do that, that's fine.